mailpoet/lib/Validator/Validator.php

```diff
@@ -25,7 +25,18 @@ public function __construct(
    * @return mixed
    */
   public function validate(Schema $schema, $value, string $paramName = 'value') {
-    $result = $this->validateAndSanitizeValueFromSchema($value, $schema->toArray(), $paramName);
+    return $this->validateArray($schema->toArray(), $value, $paramName);
+  }
+
+  /**
+   * Strict validation & sanitization implementation.
+   * It only coerces int to float (e.g. 5 to 5.0).
+   *
+   * @param mixed $value
+   * @return mixed
+   */
+  public function validateArray(array $schema, $value, string $paramName = 'value') {
```

Comment:
👍, just thinking about the naming here... Maybe this could be confused with "validating an array" rather than "validating anything but having the schema specified as array". But not sure how to name it better.

Another thing is that the `Validator` wrapper has some constraints on the JSON schema in the array, and those constraints are enforced with the typed `Schema`. So this method is sort of "unsafe" in terms of anything that can be passed into it. So maybe naming like `rawValidate` or similar would catch this better?

(On the other hand, we can remove this method, if we fully implement the new validation logic inside the validator one day.)

Reply:
Addressed in[9a11683](https://github.com/mailpoet/mailpoet/pull/4447/commits/9a11683d416d6b448902255f0abfc0beca0ffe06)